examples/LoRaWAN/LoRaWAN_ESP32/LoRaWAN_ESP32.ino

```diff
@@ -0,0 +1,196 @@
+
+/*
+
+This demonstrates how to save the join information in to permanent memory
+so that if the power fails, batteries run out or are changed, the rejoin
+is more efficient & happens sooner due to the way that LoRaWAN secures
+the join process - see the wiki for more details.
+
+This is typically useful for devices that need more power than a battery
+driven sensor - something like a air quality monitor or GPS based device that
+is likely to use up it's power source resulting in loss of the session.
+
+The relevant code is flagged with a ##### comment
+
+Saving the entire session is possible but not demonstrated here - it has
+implications for flash wearing and complications with which parts of the 
+session may have changed after an uplink. So it is assumed that the device
+is going in to deep-sleep, as below, between normal uplinks.
+
+*/
+
+#if !defined(ESP32)
+  #pragma error ("This is not the example your device is looking for - ESP32 only")
+#endif

```

Comment:
@HeadBoffin I don't think this example should be in RadioLib - it's specific to ESP32 only. Yes, ESP32 is quite specific in some regards that do have impact on LoRaWAN usage in the most common cases - most notably the fact that it does not retain RAM in deep sleep. 

Another point is there are other devices that also have their specifics, and we do not include special examples for those - for example, the LLCC68 radio that only supports some spreading factor/bandwidth combinations, and the user needs to be aware of that to disable ADR and select a valid join data rate.

What I'm specifically trying to avoid is this: https://github.com/StuartsProjects/SX12XX-LoRa/tree/master/examples/SX126x_examples Stuart has a great showcase of various LoRa projects that use cameras, GPS, SD cards etc., but the scope is so wide most of the commits are "update example" or "update readme". Even if that was the design goal of that repository (as stated in its readme), I prefer to focus on either the driver side, or the application side, but not try to cover both in one repository.

For cases with specific hardware (ESP32, LLCC68, CubeCell etc.), we can include some info/links in the wiki. Let's leave it to the community (e.g. @ropg) to provide examples/showcases for specific hardware that can be bit more wide in their scope than RadioLib.

Reply:
> Stuart has a great showcase of various LoRa projects 

Stuart is, in my book, untouchable, so I may have to kill you now for dis'ing him but then I guess you get close with Fossasat so I'll let you off just this once ;-)

All of the above said, the most prevalent boards with LoRa radios on the market are the Lilygo and Heltec offerings so a lot of new users end up with an ESP32 based board and then come unstuck when they find out that it won't work on batteries - LMIC-node being that prime example of leading people down the path to disappointment.

And low power mode is a function of a LoRaWAN device in 99% of use cases so I'd strongly recommend lots of signposting and a repro with the canon examples for people to build on based on the architecture of RadioLib's LoRaWAN. Then support issues can devolve down to "we know this works, other libraries YMMV, go talk to them".

But as a minority user of ESP32 for LW but using it rather a lot for BLE & WiFi gateways but with my eyes on a some nice combos of ESP32+SX1262 that may make it more attractive (hence all the effort), I'm fine with a readme that says go here for more examples.

Reply:
Me dis'ing Stuart is not personal, I'm dis'ing everone equally :P I'm sure he's a cool guy.

> a readme that says go here for more examples.

Either that or links in examples - fine as well.